app/models/concerns/distanceable.rb

```
@@ -12,6 +12,67 @@ def total_distance(points = nil, unit = :km)
+        end
+      end
+
+      result = total_meters.to_f / ::DISTANCE_UNITS[unit.to_sym]
+
+      # Final validation of result
+      if !result.finite?
+        return 0
+      end
+
+      result
+    end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `calculate_distance_for_array_geocoder` is too high. [<11, 37, 23> 44.93/17]

(Metrics/AbcSize)